spark/src/main/scala/org/apache/spark/sql/delta/files/DelayedCommitProtocol.scala

```diff
@@ -164,7 +183,8 @@ class DelayedCommitProtocol(
    */
   override def newTaskTempFile(
       taskContext: TaskAttemptContext, dir: Option[String], ext: String): String = {
-    val partitionValues = dir.map(parsePartitions).getOrElse(Map.empty[String, String])
+    val partitionValues = dir.map(dir => parsePartitions(dir, taskContext))
```

Comment:
Scala esthetics nit. You could use currying: 
```suggestion
    val partitionValues = dir.map(parsePartitions(taskContext))
```

And, then the signature of the parsePartitions would look like:
```scala
protected def parsePartitions(taskContext: TaskAttemptContext)(dir: String): Map[String, String] = {
```